runtime/driver/src/main/java/org/apache/nemo/driver/UserApplicationRunner.java

```diff
@@ -75,6 +87,7 @@ public synchronized void run(final String dagString) {
       LOG.info("##### Nemo Compiler Start #####");
 
       final IRDAG dag = SerializationUtils.deserialize(Base64.getDecoder().decode(dagString));
+      dag.setEnvironmentType(this.environmentType);
```

Comment:
Hmm.. is it natural that IRDag itself has an environment type?

Reply:
It's an optional field, and my thoughts were that the optimizer could simply optimize using the DAG and the environment tag on it. Actually, I could separate them out easily.